### PR TITLE
[12.0][ADD] base_model_restrict_update

### PR DESCRIPTION
**Pull request by TimLai125**
This module borrows the idea from @moylop260 in the following thread.
https://odoo-community.org/groups/contributors-15/contributors-161807

Adding update restriction config to `ir.model`, which bypass standard access right and only allows users with update permit to update/create/delete the records.

### CODE REVIEW

File: base_model_restrict_update/views/res_users_views.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    <record id="view_users_form" model="ir.ui.view">
+        <field name="name">res.users.form</field>
+        <field name="model">res.users</field>
+        <field name="inherit_id" ref="base.view_users_form" />
+        <field name="arch" type="xml">
+            <div class="oe_button_box" name="button_box" position="inside">
+                <button name="toggle_unrestrict_model_update" type="object" class="oe_stat_button" icon="fa-pencil">
+                    <field name="unrestrict_model_update" widget="boolean_button"
+                           options='{"terminology": {"string_true": "Update Permit", "string_false": "Unpermitted"}}'/>
```
Comment (yostashiro) [inline base_model_restrict_update/views/res_users_views.xml:12]:
```suggestion
                           options='{"terminology": {"string_true": "Unrestricted", "string_false": "Restricted"}}'/>
```

File: base_model_restrict_update/models/ir_model.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class IrModel(models.Model):
+    _inherit = "ir.model"
+
+    restrict_update = fields.Boolean(
+        "Update Restrict Model",
```
Comment (yostashiro) [inline base_model_restrict_update/models/ir_model.py:11]:
```suggestion
        "Restrict Update",
```

**Review by yostashiro — APPROVED** [review]
Functional review. 👍  Just a couple of non-blocking suggestions.

**Review by tosolini — APPROVED** [review]

**Pull request merged**